portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -69,5 +73,21 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     response.sendRedirect("/index.html");
   }
+
+  private int getNumComments(HttpServletRequest request) {
+    String maxCommentsString = request.getParameter("max-comments");
+    int numMaxComments;
+    try {
+      numMaxComments = Integer.parseInt(maxCommentsString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + maxCommentsString);
+      return -1;
```

Comment:
Changed the return type for getMaxComments to Integer.

For the null check in doGet, I made it so that the maxComments variable will just be set to 0 if getMaxComments gave a null value.